### DISCUSSION TITLE
All requests leading to "django.request log Bad Request"

### DISCUSSION THREAD

**Discussion by marc-leinweber** [pretix/pretix#2601] [Q&A] [answered]
Hey all,

I am trying to setup a pretix ticket shop. I decided to go with the manual deployment. I am using MySQL as DBMS as we need it for other services.

My current situation: all requests end up in a (correctly styled) bad request response and the following log output in /var/log/data/logs:
```
WARNING 2022-04-24 05:39:53,695 django.request log Bad Request: /
WARNING 2022-04-24 06:07:25,148 django.request log Bad Request: /control
```

My nginx config logs like this:
```
server {
    listen 0.0.0.0:443 ssl http2 ;
    listen [::]:443 ssl http2 ;

    server_name tickets._.de ;

    add_header X-Frame-Options "SAMEORIGIN";
    add_header Strict-Transport-Security "max-age=31536000; includeSubdomains" ;
    add_header X-Content-Type-Options "nosniff";
    add_header X-XSS-Protection "1; mode=block";

    client_max_body_size 100M ;
    client_body_timeout 60s ;

    location / {
        proxy_pass http://127.0.0.1:8081 ;
    }
    location /media/ {
      alias /var/pretix/data/media/;
      expires 7d;
      access_log off;
    }
    location ^~ /media/cachedfiles {
      deny all;
      return 404;
    }
    location ^~ /media/invoices {
      deny all;
      return 404;
    }
    location /static/ {
      alias /var/pretix/venv/lib/python3.8/site-packages/pretix/static.dist/;
      access_log off;
      expires 365d;
      add_header Cache-Control "public";
    }
}
```

Proxy and TLS config left out.

My current pretix config file looks like this:

```
[pretix]
instance_name=Kartenverkauf
url=https://tickets._.de
currency=EUR
datadir=/var/pretix/data
cookie_domain=tickets._.de
ecb_rates=off
trust_x_forwarded_for=on
trust_x_forwarded_proto=on

[locale]
default=de
timezone=Europe/Berlin

[database]
backend=mysql
name=pretix
user=pretix
...
host=/var/run/mysqld/mysqld.sock

[mail]
...

[metrics]
enabled=false

[redis]
location=redis://127.0.0.1:6379/0
sessions=true

[languages]
enabled=en,de

[celery]
backend=redis://127.0.0.1:6379/1
broker=redis://127.0.0.1:6379/2
```

And, finally, the pretix web systemd service:

```
[Unit]
Description=pretix web service
After=network.target

[Service]
User=pretix
Group=pretix
Environment="VIRTUAL_ENV=/var/pretix/venv"
Environment="PATH=/var/pretix/venv/bin:/usr/local/bin:/usr/bin:/bin"
ExecStart=/var/pretix/venv/bin/gunicorn pretix.wsgi \
                      --name pretix --workers 4 \
                      --max-requests 1200  --max-requests-jitter 50 \
                      --log-level=debug --bind=127.0.0.1:8081
WorkingDirectory=/var/pretix
Restart=on-failure

[Install]
WantedBy=multi-user.target
```

I am very confused that I do not get any useful debugging output. I had no error messages during setup (besides the ones you can't get rid of during rebuild/updatestyles).

Do you need any more information?

Thanks a lot for your help,
Marc

**Comment by raphaelm** [comment, marked as answer]
There's a 90% chance there's a mismatch between the URL set in pretix.cfg and what nginx actually sends to pretix in the Host header. Not exactly sure why, though, from a very quick(!) glance the config looks okay

**Reply by raphaelm** [reply]
Either there's a typo in the URLs you redacted, or the nginx config is wrong in ways I don't see right now, or the pretix.cfg file is not read at all, eg in the wrong path etc

**Reply by marc-leinweber** [reply]
Thank you very much for your quick answer. I'll have a look at your hints in the next days and will come back to you :)

**Reply by marc-leinweber** [reply]
Looks like my pretix config file had the wrong filename (config.cfg instead of pretix.cfg). The rest of my config was totally fine...

Thank's a lot for the hint!

**Comment by pc-coholic** [comment]
Housekeeping: Deleted the two original revisions of the issue containing passwords. @marc-leinweber, I still recommend you cycle the passwords that were exposed, as everyone subscribed to pretix issues has received the passwords in question also by mail.

**Reply by marc-leinweber** [reply]
Thank you very much. I noticed my mistake directly (copy and paste from wrong terminal)  and all passwords are already cycled :)

**Comment by choseh** [comment]
Just a small addition from my side, my location looks like this:

```
    location / {
        proxy_pass http://localhost:8345;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto https;
        proxy_set_header Host $http_host;
    }
```

maybe give this a try, if you can't find other issues with your setup.
